### ISSUE TITLE
Feature Request - Git challenges

### ISSUE THREAD

**Issue by Nirajn2311** [freeCodeCamp/curriculum#273]
#### Is your feature request related to a problem? Please describe.
I think that it will be better if FCC taught users on how to use git as many people who after learning in FCC may start making their own projects and git will be very useful as it is a powerful VCS and to make sure they don't mistakenly delete some files or folders of their projects.(I'm sure most of us have been there when programming).

#### Additional context
This feature request is opened again as the previous one was mistakenly connected to a PR.

**Comment by dougouverson** [comment]
This could be part of a broader "Developer Tools Certification (300 hours)" section?

**Comment by dougouverson** [comment]
I'm thinking it might be best to stay with subjects that would be ubiquitous in adoption; that's why I was hesitant to add VS Code. Even vim vs nano is subjective.

**Comment by RaspberryLime** [comment]
I’d vote against specific code editors. If specific code editors were to be included though, then in my opinion something which works on _at least_ Windows and Mac OS X should be used. (I’m actually trying to get through the entire course using only iOS.)

Maybe FTP could be covered in some wa too, in at least how file permissions/CHMOD work.

**Comment by dougouverson** [comment]
Yes, I agree. Developer Tools Basics: command line basics, FTP basics, git basics, etc. Subjects that are common to developers of all stripes.

**Comment by RaspberryLime** [comment]
How about adding in how to validate HTML and CSS code, using say https://validator.w3.org ? Or is this something which should be taught much earlier on? (My preference would be much earlier on.)
